source/Calamari/Scripts/Octopus.Features.Nginx_BeforePreDeploy.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Enter any commands that are required by the NGINX feature in a way the program doesn't do anything, like displaying the commands help
+requiredCommandsToCheck="cp --help
+mv --help
+rm --help
+systemctl --help
+nginx -h"
+
+function check_user_has_sudo_access_without_password_to_command {
+    sudo -n $1 $2 > /dev/null 2>&1
+    if [[ $? -ne 0 ]]; then
+        echo >&2 "User does not have 'sudo' access without password to command '$1'"
+        failedSudoCheck=1
+    fi
+}
+
+function check_user_has_sudo_access_without_password_to_required_commands {
+    IFS=$'\n' read -rd '' -a cmdArr <<< "$requiredCommandsToCheck"
+    
+    failedSudoCheck=0
+    for cmd in "${cmdArr[@]}"
+    do
+        check_user_has_sudo_access_to_command ${cmd}
```

Comment:
I think this is supposed to be `check_user_has_sudo_access_without_password_to_command`? I get command not found errors:

/home/ubuntu/.octopus/Applications/feat-nginx/Dev/dummy/0.0.1_2/Octopus.Features.Nginx_BeforePreDeploy.sh: line 24: check_user_has_sudo_access_to_command: command not found 

Interestingly the deployment continues.

Reply:
Because I don't know how to write `for`-loops in `bash`, should be:
```
for ...
do
   ...
done || exit $?
```
to exit with correct exit code if something goes unexpectedly wrong in the `for`-loop